mizar/daemon/interface_service.py

```diff
@@ -219,8 +221,51 @@ def ConsumeInterfaces(self, request, context):
         logger.error("Timeout, no new interface to consume!")
         return self._ConsumeInterfaces(requested_pod_name, cni_params)
 
+    def _UpdateSubstrateInterface(self, interface, add=True):
+        pod_name = get_pod_name(interface.interface_id.pod_id)
+        droplet_ip = interface.droplet.ip_address
+        droplet_mac = interface.droplet.mac
+        if add:
+            self._UpdateSubstrate(pod_name, droplet_ip, droplet_mac)
+        else:
+            self._DeleteSubstrate(pod_name, droplet_ip, droplet_mac)
+
+    def _UpdateSubstrate(self, obj_name, droplet_ip, droplet_mac):
+        if obj_name not in self.known_substrates.keys():
+            self.known_substrates[obj_name] = droplet_ip
+        self.rpc.update_substrate_ep(droplet_ip, droplet_mac)
+
+    def _DeleteSubstrate(self, obj_name, droplet_ip, droplet_mac):
+        if obj_name in self.known_substrates.keys():
+            self.known_substrates.pop(obj_name)
+        self.rpc.update_substrate_ep(droplet_ip, droplet_mac)
+
+    def _DeleteVethInterface(self, interface):         
+        """
+        Delete a veth interface
+        """
+        veth_peer_index = get_iface_index(interface.veth.peer, self.iproute)
+        
+        self.iproute.link('del', index=veth_peer_index)
+
+        self._UpdateSubstrateInterface(interface, False)
```

Comment:
We don't need this, it can be removed.
The transit xdp program does not need to update/delete the substrate information about its own node.